### ISSUE TITLE
trainer intermittently fails to deploy

### ISSUE THREAD

**Issue by danmelamed** [multiworldtesting/ds-provisioning#7]
error is {
  "status": "Failed",
  "error": {
    "code": "ResourceDeploymentFailure",
    "message": "The resource operation completed with terminal provisioning state 'Failed'.",
    "details": [
      {
        "code": "DeploymentFailed",
        "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details.",
        "details": [
          {
            "code": "BadRequest",
            "message": "{\r\n  \"error\": {\r\n    \"code\": \"DeploymentSlotCreateOperationFailed\",\r\n    \"message\": \"The create deployment operation failed for the domain 'DMDP6-trainer-vrjgsolb5os6y' in the deployment slot 'production' with the name 'DMDP6-trainer-vrjgsolb5os6y': 'The supplied package is invalid.'.\"\r\n  }\r\n}"
          }
        ]
      }
    ]
  }

current work-around: retry

adding milestone, since this should be at least documented, maybe in the troubleshooting docs

**Comment by eisber** [comment]
Hmmm. I tried to fix that.

**Comment by danmelamed** [comment]
Happened again.  RG = DMDP3

**Comment by danmelamed** [comment]
Just happened 3 times in a row.

**Comment by eisber** [comment]
That sucks… there’s a race condition in DeploymentController…

**Comment by lhoang29** [comment]
Yep, I think the main problem is in the copying and uploading cspkg from github to blob, this takes a while to run so it's very likely that two processes try to upload at the same time and fail. The rest of the code is idempotent. The nice thing is the settings blob already exists at this point so we can perhaps try to obtain a lease on it.

**Comment by lhoang29** [comment]
wait, upon closer look this is entirely unrelated issue. the problem is the underscores in VM size are removed, these need to stay otherwise they are invalid :). I'm going to put a fix together

**Comment by danmelamed** [comment]
Ooh, that's my bad.  I can fix it.

**Comment by lhoang29** [comment]
I already did, and after testing it looks like the deployment is passing again now (twice for me)
